### PR TITLE
Shooting and Animations

### PR DESCRIPTION
**Pull request by cnnradams**
Shoot with Space and Animations with a pacman lol

### CODE REVIEW

File: Luminescent/.classpath [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	<classpathentry kind="lib" path="libs/lwjgl/lwjgl.jar" sourcepath="libs/lwjgl/lwjgl-src.zip">
 		<attributes>
 			<attribute name="javadoc_location" value="jar:platform:/resource/Luminescent/libs/lwjgl/lwjgl-doc.zip!/"/>
-			<attribute name="org.eclipse.jdt.launching.CLASSPATH_ATTR_LIBRARY_PATH_ENTRY" value="Luminescent/libs/lwjgl/natives"/>
+			<attribute name="org.eclipse.jdt.launching.CLASSPATH_ATTR_LIBRARY_PATH_ENTRY" value="C:/Users/cnnr2/Desktop/lwjgl/native"/>
```
Comment (ryanlarkin) [inline Luminescent/.classpath]: Can you fix this?

File: Luminescent/src/astechzgo/luminescent/main/Luminescent.java [hunk 2]
```diff
@@ -95,34 +90,103 @@ public static void Tick()
...
+			
+			if(thePlayer.getPosY() - speed * Math.cos(Math.toRadians(angle)) >= room.getPosY() + room.getHeight() - thePlayer.getRadius())
+				thePlayer.setPosY(room.getPosY() + room.getHeight() - thePlayer.getRadius());
+			else if(thePlayer.getPosY() - speed * Math.cos(Math.toRadians(angle)) <= room.getPosY() + thePlayer.getRadius())
+				thePlayer.setPosY(room.getPosY() + thePlayer.getRadius());
+			else
+				thePlayer.setPosY(thePlayer.getPosY() - speed * Math.cos(Math.toRadians(angle)));
 		}
+		if(KeyboardUtils.isKeyDown(Constants.KEYS_ACTION_SHOOT)) {
+			System.out.println("crash1");
```
Comment (ryanlarkin) [inline Luminescent/src/astechzgo/luminescent/main/Luminescent.java]: Can you also get rid of the System.out calls?

File: Luminescent/src/astechzgo/luminescent/rendering/OpenGL.java [hunk 3]
```diff
@@ -34,6 +34,8 @@ public static void ChangeResolution(int x, int y)
 		GL11.glLoadIdentity();
 		GL11.glOrtho(0, x, 0, y, 1, -1);
 		GL11.glMatrixMode(GL11.GL_MODELVIEW);
+		GL11.glEnable(GL11.GL_BLEND);
+	    GL11.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA);
```
Comment (ryanlarkin) [inline Luminescent/src/astechzgo/luminescent/rendering/OpenGL.java]: Formatting

File: Luminescent/src/astechzgo/luminescent/rendering/Projectile.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import astechzgo.luminescent.main.Luminescent;
+
+public class Projectile extends RenderableRectangularGameObject {
+	private double lastDelta;
+	private double rotation;
+	public Projectile(int x, int y) {
+		super(x, y, 5,5);
+		super.setColour(new Color(0.5f, 0.6f, 0.2f));
+		 lastDelta = GLFW.glfwGetTime() * 1000;
+	rotation = Luminescent.thePlayer.setRotation();
```
Comment (ryanlarkin) [inline Luminescent/src/astechzgo/luminescent/rendering/Projectile.java]: Formatting

File: Luminescent/src/astechzgo/luminescent/rendering/Projectile.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	}
+	
+public void fireBullet() {
+	double delta = ((GLFW.glfwGetTime() * 1000) - lastDelta);
+	super.x = (int) (Luminescent.thePlayer.getPosX() + 1 * delta* Math.cos(Math.toRadians(-rotation)));
+	super.y = (int) (Luminescent.thePlayer.getPosY() + 1 * delta *Math.sin(Math.toRadians(-rotation)));
```
Comment (eandr127) [inline Luminescent/src/astechzgo/luminescent/rendering/Projectile.java]: You can just say x.  I'll fix this during cleanup.

File: Luminescent/src/astechzgo/luminescent/utils/Constants.java [hunk 6]
```diff
@@ -33,11 +35,15 @@
...
 		defaults.put(LOG_CONFIG, "false");
 		
+
+		
 		defaults.put(KEYS_MOVEMENT_FASTER,				"{LEFT_SHIFT}");
 		defaults.put(KEYS_MOVEMENT_UP, 					"{W}, {UP}");
 		defaults.put(KEYS_MOVEMENT_LEFT, 				"{A}, {LEFT}");
 		defaults.put(KEYS_MOVEMENT_DOWN, 				"{S}, {DOWN}");
-		defaults.put(KEYS_MOVEMENT_RIGHT, 				"{D}, {RIGHT}");
+		defaults.put(KEYS_MOVEMENT_RIGHT,               "{D}, {RIGHT}");
```
Comment (ryanlarkin) [inline Luminescent/src/astechzgo/luminescent/utils/Constants.java]: Formatting

File: Luminescent/src/astechzgo/luminescent/utils/DisplayUtils.java [hunk 7]
```diff
@@ -211,9 +211,8 @@ public static void setDisplayMode(int width, int height, boolean fullscreen) {
 			GL11.glLoadIdentity();
 			GL11.glOrtho(0, mode.WIDTH, 0, mode.HEIGHT, 1, -1);
 			GL11.glMatrixMode(GL11.GL_MODELVIEW);
-			
-			if(fullscreen)
-				GLFW.glfwSetInputMode(handle, GLFW.GLFW_CURSOR, GLFW.GLFW_CURSOR_DISABLED);
+			GL11.glEnable(GL11.GL_BLEND);
```
Comment (eandr127) [inline Luminescent/src/astechzgo/luminescent/utils/DisplayUtils.java]: Formatting

**Comment by ryanlarkin** [comment]
Do you have any issues with this?

**Pull request merged**